compressors/zstd/src/main/java/io/opentelemetry/contrib/compressor/zstd/ZstdCompressor.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.compressor.zstd;
+
+import com.github.luben.zstd.ZstdOutputStream;
+import io.opentelemetry.exporter.internal.compression.Compressor;
+import java.io.IOException;
+import java.io.OutputStream;
+
+public final class ZstdCompressor implements Compressor {
+
+  private static final ZstdCompressor INSTANCE = new ZstdCompressor();
+
+  private ZstdCompressor() {}
+
+  public static ZstdCompressor getInstance() {
+    return INSTANCE;
+  }
+
+  @Override
+  public String getEncoding() {
+    return "zstd";
+  }
```

Comment:
(unrelated to this PR) was there reason for this method to be in `Compressor` and not `CompressorProvider` (e.g. similar to `ConfigurableSpanExporterProvider.getName()`)

Reply:
Yeah so this is a good question which I encountered implementing. My thinking:
- We want some sort of name that we can reference when we call `setCompression(String)` on the exporters
- We also want a value to use for the `Content-Encoding` header
- I think it will always be sufficient to use the same name for `setCompression` and `Content-Encoding`.

By the time the Compressor is resolved from CompressorProvider, the compressor wouldn't be able to access a `CompressorProvider#getName` any more. So my thinking was, put the identifier down in Compressor instead of CompressorProvider, so its always accessible to the Compressor where its needed for `Content-Encoding`. Count on the fact that `CompressorProvider#getInstance()` _should_ return singletons, and therefore you can call `CompressorProvider.getInstance().getName()` to get the name identifier of the `CompressorProvider`.

I also thought about how [ContextStorageProvider](https://github.com/open-telemetry/opentelemetry-java/blob/main/context/src/main/java/io/opentelemetry/context/ContextStorageProvider.java) also doesn't have a `#getName()` method, since its identified by its FQCN for configuration. Compressor is different because we definitely want to be able to use a short name instead of FQCN, but that's a little bit of the prior art and what I was thinking about.